### PR TITLE
fix some bugs in BKTree and SPANNIndex, add new result format for indexsearcher

### PR DESCRIPTION
**Pull request by MaggieQi**

### CODE REVIEW

File: AnnService/src/IndexSearcher/main.cpp [hunk 1]
```diff
@@ -69,6 +70,8 @@ class SearcherOptions : public Helper::ReaderOptions
     int m_debugQuery = -1;
 
     bool m_enableADC = false;
+
+    int m_outputformat = 0;
```
Comment (PhilipBAdams) [inline AnnService/src/IndexSearcher/main.cpp:74]: Can we use SPTAG::VectorFileType instead

File: AnnService/src/IndexSearcher/main.cpp [hunk 2]
```diff
@@ -237,37 +254,76 @@ int Process(std::shared_ptr<SearcherOptions> options, VectorIndex& index)
...
+                for (SizeType i = 0; i < numQuerys; i++)
                 {
-                    if (results[i].GetResult(j)->VID < 0) {
-                        fp << results[i].GetResult(j)->Dist << "@NULL" << std::endl;
-                        continue;
-                    }
-
-                    if (!options->m_withMeta) {
-                        fp << (results[i].GetResult(j)->Dist / baseSquare) << "@" << results[i].GetResult(j)->VID << std::endl;
+                    if (queryMetas != nullptr) {
```
Comment (PhilipBAdams) [inline AnnService/src/IndexSearcher/main.cpp:262]: If we want to support multiple output formats, should we wrap it in a `VectorSetWriter` the way we do with multiple input formats? I think we have this kind of logic in SSDServing too

**Review by PhilipBAdams — APPROVED** [review]

File: AnnService/src/IndexSearcher/main.cpp (hunk 1 above)
Comment (MaggieQi) [reply AnnService/src/IndexSearcher/main.cpp:74]: It is the result output type not the input vector type

File: AnnService/src/IndexSearcher/main.cpp (hunk 2 above)
Comment (MaggieQi) [reply AnnService/src/IndexSearcher/main.cpp:262]: Yes, good suggestion! We can do it after this pull request.

**Pull request merged**